pipelinedp4j/main/com/google/privacy/differentialprivacy/pipelinedp4j/spark/SparkTable.kt

```diff
@@ -149,7 +167,40 @@ class SparkTable<K, V>(val data: Dataset<Tuple2<K, V>>,
         return filterKeys(stageName) {k -> k in allowedKeysHashSet}
     }
 
+    /**
+     * Randomly samples values per key. The output table will contain same keys as this table, each key will appear
+     * only once. The number of values per key will be at most [count].
+     * It uses window partition by function which requires an extra shuffle and sort operation and introduces
+     * an extra step to transfer data over network but is a scalable and efficient approach for large dataset.
+     */
     override fun samplePerKey(stageName: String, count: Int): SparkTable<K, List<V>> {
```

Comment:
hm, does it compile? The interface returns `Iterable`: `FrameworkTable<K, Iterable<V>>`.